File: java-workspace2/src/프로그래머스/소수만들기.java
package 프로그래머스;

public class 소수만들기 {
    static int answer = 0;
    public static void main(String[] args) {
        solution(new int[]{1,2,7,6,4});
    }
    public static int solution(int[] nums) {
        dfs(nums, 0, 0, 0 );
        System.out.println(answer);
        return answer;
    }
    private static void dfs(int[] nums, int idx, int count, int sum) {
        if(count == 3) {
            if(check(sum)){
                answer++;
            }
            return;
        }
        if(idx ==nums.length){
            return;
        }
        dfs(nums, idx+1, count+1, sum+nums[idx]);
        dfs(nums, idx+1, count, sum);
    }
    private static boolean check(int num) {
        for(int i = 2 ; i<num; i++) {
            if(num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: python-workspace/완전탐색/소수만들기.py
from itertools import combinations

def solution(nums):
    comb = combinations(nums, 3)
    count = 0
    for c in comb:
        if is_prime_number(sum(c)):
            count += 1
    return count


def is_prime_number(sum):
    if sum == 1:
        return False
    if sum == 2:
        return True
    for num in range(2, sum):
        if sum % num == 0:
            return False
    return True


solution([1, 2, 3, 4])
